exatomic/gaussian/output.py

```diff
@@ -608,6 +619,212 @@ def parse_momatrix(self):
         if bcoefs is not None:
             self.momatrix['coef1'] = bcoefs
 
+    def parse_frequency_ext(self):
+        '''
+        Parses extended frequency data present in FChk file
+
+        Note:
+            Requires Freq(SaveNormalModes) in input
+        '''
+        # Frequency regex
+        _renmode = 'Number of Normal Modes'
+        _redisp = 'Vib-Modes'
+        _refinfo = 'Vib-E2'
+        _remass = 'Vib-AtMass'
+        # TODO: find out what these two values are useful for
+        _rendim = 'Vib-NDim'
+        _rendim0 = 'Vib-NDim0'
+
+        found = self.find(_renmode)
+        if not found:
+            return
+        else:
+            found = self.find(_renmode, _redisp, _refinfo, _remass, keys_only=True)
+        # find number of vibrational modes
+        nmode = self._intme(found[_renmode])
+        # get extended data (given by mapper array)
+        all_info = self._dfme(found[_refinfo], nmode * 14)
```

Comment:
I think this parsing logic is a bit fragile. If any of the strings (e.g. `_renmode`) are not found a `KeyError` will be thrown. I think it might be worth refactoring this parsing logic to early exit if a key is missing. See also the above comment.

Reply:
I'll look into it, but from what I understand with Gaussian the frequencies will only be written to the FChk file when we pass Freq=(SaveNormalModes) this will print out some data that have a very well defined number points. When it is not present it will not find anything for any of the keys. Again I'll make absolutely sure of this by running some small examples and see if I get any errors.